### PR TITLE
feat: [CDS-70060]: Enabled GitHub package registry artifact for ssh/winrm

### PR DESCRIPTION
**Pull request by litvinovstas**
### Summary
Enabled GitHub package registry artifact for ssh/winrm

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactsSelection.tsx [hunk 1]
```diff
@@ -117,6 +118,11 @@ export default function ArtifactsSelection({
     ) {
       allowedArtifactTypes[deploymentType].push(ENABLED_ARTIFACT_TYPES.Bamboo)
     }
+    if (!CDS_GITHUB_PACKAGES && isSshOrWinrmDeploymentType(deploymentType)) {
+      allowedArtifactTypes[deploymentType].filter(
```
Comment (ishantmehtaharness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactsSelection.tsx:122]: Here, we are changing `allowedArtifactTypes`, which is imported from another file
We should not change it, we should ideally create another variable

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactsSelection.tsx [hunk 2]
```diff
@@ -117,6 +118,11 @@ export default function ArtifactsSelection({
     ) {
       allowedArtifactTypes[deploymentType].push(ENABLED_ARTIFACT_TYPES.Bamboo)
```
Comment (ishantmehtaharness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactsSelection.tsx:119]: Same for this also
https://github.com/harness/harness-core-ui/pull/16823/files#r1259935318

**Review by satyam-harness — DISMISSED** [review]

**Review by satyam-harness — APPROVED** [review]

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactsSelection.tsx [hunk 3]
```diff
@@ -99,23 +100,28 @@ export default function ArtifactsSelection({
   const [context, setModalContext] = useState(ModalViewFor.PRIMARY)
   const [sidecarIndex, setEditIndex] = useState(0)
   const [fetchedConnectorResponse, setFetchedConnectorResponse] = useState<PageConnectorResponse | undefined>()
-
+  const [artifactTypes, setArtifactTypes] = useState<ArtifactType[]>(allowedArtifactTypes[deploymentType])
```
Comment (ishantmehtaharness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactsSelection.tsx:103]: Thank You Stan for making the changes
Can you verify if this works when you change the deployment type

**Review by ishantmehtaharness — APPROVED** [review]
One comment
Please verify that before you merge

https://github.com/harness/harness-core-ui/pull/16823#discussion_r1260668149

**Pull request merged**